docs/source/services.md

```diff
@@ -0,0 +1,84 @@
+# JupyterHub services
+
+JupyterHub 0.7 adds the notion of Services.
+A Service is a process that interacts with the Hub REST API.
+Services may perform actions such as shutting down user servers that have been idle for some time,
+or registering additional web servers that should also use the Hub's authentication
+and be served behind the Hub's proxy.
+
+There are two main characteristics of services:
+
+1. Is it **managed** by JupyterHub?
+2. Does it have a web server that should be added to the proxy?
+
+If a `command` is specified for launching the service, it will be started and managed by the Hub.
+If a `url` is specified for where the service runs its own webserver,
+it will be added to the Hub's proxy at `/service/:service-name`.
+
+## Managed services
+
+**Managed** services are services that the Hub starts and is responsible for.
+These can only be local subprocesses of the Hub.
+While there are similarities with notebook Spawners,
+there are no plans to support the same spawning abstractions as notebook
+If you want to run these services in docker or other environments,
```

Comment:
I think you are right, but I'll clarify a few things to be sure.

There's two use cases I intend to cover here:

1. simple subprocesses run locally, managed by the Hub
2. externally managed services, where this code only helps with:
    1. setting up API tokens (e.g. cull-idle, HubAuth)
    2. setting up proxy forwarding for web services (e.g. formgrader)

If what you are after is keeping an external service up while the Hub is up, then probably adding it one level up at the docker-compose (or similar) level, rather than internal to JupyterHub, makes the most sense to me.